includes: remove duplicated entries in zephyr-tree

sylvioalves opened the pull request:
Remove duplicated #include directives within the same preprocessor scope across the Zephyr tree.

Duplicates inside different #ifdef branches are preserved as they may be intentional.

sylvioalves committed:
includes: remove duplicated entries in zephyr-tree

Remove duplicated #include directives within the same
preprocessor scope across the Zephyr tree.

Duplicates inside different #ifdef branches are preserved
as they may be intentional.

Signed-off-by: Sylvio Alves <sylvio.alves@espressif.com>

Thalley reviewed (dismissed):
Only reviewed BT related changes, but LGTM. A few minor comments

Thalley commented:
@jhedberg I'm pretty sure someone else than I should be the assignee. Who should I/we assign?

sylvioalves committed:
drivers: flash: esp32: preserve volatile qualifier in IPM callback

Preserve the volatile qualifier when casting the shared memory pointer
in flash_cpu01_receive_cb() to avoid dropping volatile semantics during
inter-processor communication.

Signed-off-by: Sylvio Alves <sylvio.alves@espressif.com>

sylvioalves commented:
sonarqube issue sounds false-positive as it is common Zephyr code.

<img width="647" height="197" alt="image" src="https://github.com/user-attachments/assets/6c98bf09-ad88-4dec-8efc-5d2b4f301706" />

kartben reviewed (commented):
commit "drivers: flash: esp32: preserve volatile qualifier in IPM callback" should be dropped?

aescolar commented on modules/hal_nordic/nrf_802154/serialization/platform/nrf_802154_spinel_backend_ipc.c:
```diff
@@ -21,7 +21,6 @@
 #include "../../src/include/nrf_802154_spinel.h"
 
 #if defined(CONFIG_SOC_NRF5340_CPUAPP)
```
If you could also remove the empty ifdef that would be nice

aescolar commented on drivers/sensor/st/lsm6dso/lsm6dso.h:
```diff
@@ -21,7 +21,6 @@
 
 #if DT_HAS_COMPAT_ON_BUS_STATUS_OKAY(st_lsm6dso, spi) || \
 	DT_HAS_COMPAT_ON_BUS_STATUS_OKAY(st_lsm6dso32, spi)
```
It would be nice to remove the empty ifdef

aescolar reviewed (approved):
+1
